crates/bevy_pbr/src/light_probe/mod.rs

```diff
@@ -0,0 +1,389 @@
+//! Light probes for baked global illumination.
+
+use bevy_app::{App, Plugin};
+use bevy_asset::load_internal_asset;
+use bevy_core_pipeline::core_3d::Camera3d;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    component::Component,
+    entity::Entity,
+    query::With,
+    reflect::ReflectComponent,
+    schedule::IntoSystemConfigs,
+    system::{Commands, Local, Query, Res, ResMut, Resource},
+};
+use bevy_math::{Affine3A, Mat4, Vec3A, Vec4};
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
+use bevy_render::{
+    extract_instances::ExtractInstancesPlugin,
+    primitives::{Aabb, Frustum},
+    render_asset::RenderAssets,
+    render_resource::{DynamicUniformBuffer, Shader, ShaderType},
+    renderer::{RenderDevice, RenderQueue},
+    texture::Image,
+    Extract, ExtractSchedule, Render, RenderApp, RenderSet,
+};
+use bevy_transform::prelude::GlobalTransform;
+use bevy_utils::{EntityHashMap, FloatOrd};
+
+use crate::light_probe::environment_map::{
+    EnvironmentMapIds, EnvironmentMapLight, RenderViewEnvironmentMaps,
+    ENVIRONMENT_MAP_SHADER_HANDLE,
+};
+
+pub mod environment_map;
+
+/// The maximum number of reflection probes that each view will consider.
+///
+/// Because the fragment shader does a linear search through the list for each
+/// fragment, this number needs to be relatively small.
+pub const MAX_VIEW_REFLECTION_PROBES: usize = 8;
+
+/// Adds support for light probes: cuboid bounding regions that apply global
+/// illumination to objects within them.
+///
+/// This also adds support for view environment maps: diffuse and specular
+/// cubemaps applied to all objects that a view renders.
+pub struct LightProbePlugin;
+
+/// A cuboid region that provides global illumination to all fragments inside it.
```

Comment:
Nit: Start the doc comment with something like "a marker component for a light probe, which is a ..."